django-backend/fecfiler/user/views.py

```diff
@@ -1,22 +1,31 @@
-from rest_framework import viewsets
-from rest_framework.decorators import action
-from .serializers import CurrentUserSerializer
-from django.http import HttpResponseBadRequest
 from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from .serializers import CurrentUserSerializer, session_security_consented_key
+from datetime import date, timedelta
 import logging
 
 logger = logging.getLogger(__name__)
 
 
-class UserViewSet(viewsets.GenericViewSet):
-    @action(detail=False, methods=["GET", "PUT"], serializer_class=CurrentUserSerializer)
-    def current(self, request):
-        if request.method == 'GET':
-            serializer = self.get_serializer(request.user)
-            return Response(serializer.data)
-        if request.method == 'PUT':
-            serializer = self.get_serializer(request.user, data=request.data)
-            if serializer.is_valid():
-                serializer.save()
-                return Response(serializer.data)
-            return HttpResponseBadRequest()
+class UserViewSet(ModelViewSet):
```

Comment:
I think ModelViewSet has more mixins than we want for the user view.  The old had the scope narrowed to CRU without Delete or List.  You can add them in manually
```
class ModelViewSet(mixins.CreateModelMixin,
                   mixins.RetrieveModelMixin,
                   mixins.UpdateModelMixin,
                   mixins.DestroyModelMixin,
                   mixins.ListModelMixin,
                   GenericViewSet):
```

Reply:
good catch! removed ModelViewSet and used just GenericViewSet with the use of actions